pkg/raft/raft.go

```diff
@@ -418,8 +428,10 @@ func newRaft(c *Config) *raft {
 		disableConfChangeValidation: c.DisableConfChangeValidation,
 		stepDownOnRemoval:           c.StepDownOnRemoval,
 	}
-
 	lastID := r.raftLog.lastEntryID()
+	// Invariant: accTerm needs to be initialized as the term of last raft entry
+	// to make sure restarted nodes can advance commit index immediately.
+	r.accTerm = lastID.term
```

Comment:
Note for reviewers. There is a case when the commit index won't be able to advance by a heartbeat message right after a restart, specifically when the `lastID.term` < `HardState.Term`. This is okay though, because the next `MsgApp` message from the `Term` leader (which is guaranteed, because this follower is behind the leader on the log) will bump `accTerm` to the leader's term, and advance the commit index.

For complete snappiness, we need #122446, i.e. persist `accTerm` to the `HardState` - this will make the "log is consistent with the leader" signal survive restarts. We need this `accTerm` for other reasons too.